### PR TITLE
simplify gas units internally

### PR DESCRIPTION
**Pull request by oxade**
NFC which uses more of the types & units from upstream
There will be continuous improvements but the goal with these units is to make math more consistent.

### CODE REVIEW

File: crates/sui-types/src/gas.rs [hunk 1]
```diff
@@ -185,24 +208,22 @@ impl<'a> SuiGasStatus<'a> {
...
     pub fn charge_consensus(&mut self) -> Result<(), ExecutionError> {
         self.deduct_computation_cost(&INIT_SUI_COST_TABLE.consensus_cost)
     }
 
     pub fn charge_publish_package(&mut self, size: usize) -> Result<(), ExecutionError> {
-        let computation_cost = INIT_SUI_COST_TABLE
-            .package_publish_per_byte_cost
-            .with_size(size);
+        let computation_cost =
+            NumBytes::new(size as u64).mul(INIT_SUI_COST_TABLE.package_publish_per_byte_cost.0);
```
Comment (emmazzz) [inline crates/sui-types/src/gas.rs:220]: Should we use deref here instead of `.0`? and some other places

Comment (oxade) [reply crates/sui-types/src/gas.rs:220]: Yes we can. Brb

Comment (oxade) [reply crates/sui-types/src/gas.rs]: Done @emmazzz

**Review by emmazzz — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**